### PR TITLE
Least authority remediation

### PR DESCRIPTION
**Pull request by trbritt**
This request addresses a few key concerns highlighted by the audit. These involve test coverage of edge cases in the higher level field extensions, an improved method for generating random elements in $\mathbb{G}_1$ and $\mathbb{G}_2$, as well as clarifying miscallaneous typos, incorrect references in the comments, etc. Additionally, suggested improvements have been included. See the official response to the audit for more details.

### CODE REVIEW

**Review by merolish — COMMENTED** [review]
As far as various corrections and changes go, nothing stands out to me as an issue.

File: src/fields/fp12.rs [hunk 1]
```diff
@@ -623,53 +606,14 @@ mod tests {
...
-                [1, 0, 0, 0],
-                [0, 6, 0, 0],
-                [5, 0, 0, 0],
-                [0, 0, 0, 4],
-                [0, 0, 3, 0],
-                [0, 2, 0, 0],
-                [1, 0, 0, 0],
-            ]);
+            let a = Fp12::rand(&mut OsRng);
+            let b = Fp12::rand(&mut OsRng);
```
[the remark this answers is not in the document]
Comment (0xAlcibiades) [reply src/fields/fp12.rs:610]: Yes, they want fuzz/property tests. This is correct IMO.

File: src/fields/extensions.rs [hunk 2]
```diff
@@ -187,7 +187,9 @@ impl<const D: usize, const N: usize, F: FieldExtensionTrait<D, N>> Default
 {
     /// Returns the default value for the field extension (all zero elements).
     fn default() -> Self {
-        Self::new(&[F::default(); N])
+        let mut retval = [F::zero(); N];
+        retval[0] = F::default();
+        Self::new(&retval)
```
[the remark this answers is not in the document]
Comment (0xAlcibiades) [reply src/fields/extensions.rs:192]: @trbritt

**Review by 0xAlcibiades — APPROVED** [review]
@trbritt please add the audit report here ser and then merge! Also, let's run the coverage report.
